spec/requests/api/v1/participant_declarations_spec.rb

```diff
@@ -375,6 +375,87 @@ def build_params(attributes)
           expect(fake_logger).to have_received(:info).with(instance_of(Array)).ordered
         end
       end
+
+      context "when existing declaration" do
+        context "has state awaiting_clawback" do
+          let!(:existing_participant_declaration) do
+            create(
+              :participant_declaration,
+              :awaiting_clawback,
+              user: ect_profile.user,
+              cpd_lead_provider:,
+              participant_profile: ect_profile,
+              course_identifier: valid_params[:course_identifier],
+            )
+          end
+
+          it "returns 200" do
+            expect(ParticipantDeclaration.count).to eq(1)
```

Comment:
To add an extra layer of assurance you always want to be a bit more precise in those types of assertions and you achieve those two lines in one assertion as well:

```ruby
expect(ect_profile.participant_profiles.awaiting_clawback.count).to eq(1)

Reply:
Those could be passing but not not for the reason you think, another factory may have created that record as a side effect, which happens a lot with FactoryBot when you're not careful.